gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -197,8 +225,17 @@ private Map<String, BoundField> getBoundFields(Gson context, TypeToken<?> type,
         if (!serialize && !deserialize) {
           continue;
         }
+        // The accessor method is only used for records. If the type is a record, we will read out values
+        // via its accessor method instead of via reflection. This way we will bypass the accessible restrictions
+        Method accessor = null;
+        if (isRecord) {
```

Comment:
By using `GsonBuilder.excludeFieldsWithModifiers` a user might include `static` fields for serialization. That would break this logic in case a record class declares a `static` field.

Maybe it would be possible to handle this with special-casing, but personally I think it would also be fine and easier to just skip `static` fields here for now if this is a record class. If necessary `static` field support could be added in a later PR, though I assume such use cases are rather rare.

Reply:
Would it not be better to just not set the accessor if the field is static. I am assuming here, but I think if the field is static, then normal reflection to write to the field would still work. It is only reflective writes to record fields that are constrained.